source/slang/slang-emit-c-like.h

```diff
@@ -398,17 +395,30 @@ class CLikeSourceEmitter: public SourceEmitterBase
 
     void executeEmitActions(List<EmitAction> const& actions);
 
+        // Emits front matter including prelude
+        // Doesn't emit generated function/types that's handled by emitGeneratedBuiltinDeclarations
+        // 
+        // We are going to emit the target-language-specific directives
+        // needed to get the default matrix layout to match what was requested
+        // for the given target.
+        //
+        // Note: we do not rely on the defaults for the target language,
+        // because a user could take the HLSL/GLSL generated by Slang and pass
+        // it to another compiler with non-default options specified on
+        // the command line, leading to all kinds of trouble.
+        //
+        // TODO: We need an approach to "global" layout directives that will work
+        // in the presence of multiple modules. If modules A and B were each
+        // compiled with different assumptions about how layout is performed,
+        // then types/variables defined in those modules should be emitted in
+        // a way that is consistent with that layout...
```

Comment:
I'm not sure the pre-existing comment is as useful here on the declaration, just because it is diving down into the guts of the particulars of what some of the front-matter *is*.